clang/lib/AST/DeclCXX.cpp

```diff
@@ -2056,40 +2056,40 @@ void CXXRecordDecl::completeDefinition() {
   completeDefinition(nullptr);
 }
 
+static bool hasPureVirtualFinalOverrider(
+    const CXXRecordDecl &RD, const CXXFinalOverriderMap *FinalOverriders) {
+  auto ExistsIn = [](const CXXFinalOverriderMap &FinalOverriders) {
+    for (const auto &[_, M] : FinalOverriders) {
+      for (const auto &[_, SO] : M) {
```

Comment:
While we're at it, can we use more descriptive names than `M` and `SO`? I'm not even sure what the latter means. I also have reservations towards `auto` here.

Reply:
Yeah, maybe not using structured bindings here would make this more legible...

Reply:
Updated the code with more descriptive names and removed structured bindings for the outer loop.

Regarding the inner loop, I found that the type of the loop variable cannot be spelled with the public APIs of the class `OverridingMethods` we are iterating through, so I kept that structured binding. But, its readability improved with the updated name. Please let me know if you have further suggestions.